RUFAS/biophysical/animal/reproduction/reproduction.py

```diff
@@ -1,2 +1,2768 @@
+            heifer_reproduction_sub_program: HEIFER_REPRODUCTION_SUB_PROTOCOLS = None,
+            cow_reproduction_program: CowReproductionProtocol = None,
+            cow_presynch_program: CowPreSynchSubProtocol = None,
+            cow_ovsynch_program: CowTAISubProtocol = None,
+            cow_resynch_program: CowReSynchSubProtocol = None,
+            ai_day: int = 0,
+            estrus_day: int = 0,
+            abortion_day: int = 0,
+            breeding_to_preg_time: int = 0,
+            conception_rate: float = 0.0,
+            cow_TAI_conception_rate: float = 0.0,
+            num_conception_rate_decreases: int = 0,
+            hormone_schedule: dict[int, dict[str, Any]] = None,
+            gestation_length: int = 0,
+            conceptus_weight: float = 0.0,
+            calf_birth_weight: float = 0.0,
+            calves: int = 0,
+            calving_interval: int = 0,
+            calving_interval_history: list[int] = None,
+            body_weight_at_calving: float = 0.0,
+            do_not_breed: bool = None,
+            estrus_count: int = 0
+    ) -> None:
+        self.heifer_reproduction_program = HeiferReproductionProtocol(AnimalConfig.heifer_reproduction_program) \
+            if heifer_reproduction_program is None else heifer_reproduction_program
+        self.heifer_reproduction_sub_program = AnimalConfig.heifer_reproduction_sub_program \
+            if heifer_reproduction_sub_program is None else heifer_reproduction_sub_program
+        self.cow_reproduction_program = CowReproductionProtocol(AnimalConfig.cow_reproduction_program) \
+            if cow_reproduction_program is None else cow_reproduction_program
+        self.cow_presynch_program = CowPreSynchSubProtocol(AnimalConfig.cow_presynch_method) \
+            if cow_presynch_program is None else cow_presynch_program
+        self.cow_ovsynch_program = CowTAISubProtocol(AnimalConfig.cow_tai_method) \
+            if cow_ovsynch_program is None else cow_ovsynch_program
+        self.cow_resynch_program = CowReSynchSubProtocol(AnimalConfig.cow_resynch_method) \
+            if cow_resynch_program is None else cow_resynch_program
+
+        self.ai_day = ai_day if ai_day else 0
+        self.estrus_day = estrus_day if estrus_day else 0
+        self.abortion_day = abortion_day if abortion_day else 0
+        self.breeding_to_preg_time = breeding_to_preg_time if breeding_to_preg_time else 0
+        self.gestation_length = gestation_length if gestation_length else 0
+
+        self.conceptus_weight = conceptus_weight if conceptus_weight else 0.0
+        self.calf_birth_weight = calf_birth_weight if calf_birth_weight else 0.0
+        self.body_weight_at_calving = body_weight_at_calving if body_weight_at_calving else 0.0
+
+        self.conception_rate = conception_rate if conception_rate else 0.0
+        self.TAI_conception_rate = cow_TAI_conception_rate if cow_TAI_conception_rate else 0.0
```

Comment:
```suggestion
        self.cow_TAI_conception_rate = cow_TAI_conception_rate if cow_TAI_conception_rate else 0.0
```
Not sure if this is intentional but it would make sense to keep it consistent if possible.

Reply:
I believe this value is also used for heifers. But it won't be passed in as an input for hefiers, rather dynamically calculated later in the process.

Reply:
Hmm, what do you think about renaming it in the docstrings to the more generic version and describing that it's used by cows and heifers? Would that be helpful to users?